tests/sqlparser_snowflake.rs

```diff
@@ -24,6 +41,120 @@ fn test_snowflake_create_table() {
     }
 }
 
+fn get_from_section_from_select_query(query: &str) -> Vec<TableWithJoins> {
+    let statement = snowflake().parse_sql_statements(query).unwrap()[0].clone();
+
+    let query = match statement {
+        Statement::Query(query) => query,
+        _ => panic!("Not a query"),
+    };
+
+    let select = match query.body {
+        SetExpr::Select(select) => select,
+        _ => panic!("not a select query"),
+    };
+
+    select.from.clone()
+}
+
+#[test]
+fn test_sf_derives_single_table_in_parenthesis() {
+    let from = get_from_section_from_select_query("SELECT * FROM (((SELECT 1) AS t))");
+
+    assert_eq!(
+        from[0].relation,
+        TableFactor::Derived {
+            lateral: false,
+            subquery: Box::new(snowflake().verified_query("SELECT 1")),
+            alias: Some(TableAlias {
+                name: "t".into(),
+                columns: vec![],
+            })
+        }
+    );
+}
+
+#[test]
+fn test_single_table_in_parenthesis() {
+    //Parenthesized table names are non-standard, but supported in Snowflake SQL
+    let from = get_from_section_from_select_query("SELECT * FROM (a NATURAL JOIN (b))");
+
+    assert_eq!(from[0].relation, nest!(table("a", None), table("b", None)));
```

Comment:
Better use `only(select.from)`, as `from[0]` doesn't check there are no other items in the FROM.

Reply:
Done